sopel/modules/dice.py

```diff
@@ -140,39 +133,50 @@ def _roll_dice(bot, dice_expression):
 
     # Dice can't have zero or a negative number of sides.
     if dice_type <= 0:
-        bot.reply("I don't have any dice with %d sides. =(" % dice_type)
-        return None  # Signal there was a problem
+        raise ValueError("I don't have any dice with %d sides. =(" % dice_type)
```

Comment:
Hmm, something feels slightly gross about "I" in an exception message, here. I understand that the intent is for this message to propagate to the user, but it feels a bit off that the exception carries this payload, in a way that I have a hard time putting my finger on.

Reply:
You're gonna laugh… This nit sent me on another refactor of the error handling, touching even more edge cases, and now at 99% file coverage with just one missed line.

But the exceptions no longer carry the message, only data _about_ the error. 😄

Reply:
It's OK, these are very edge cases, and people using IRC are used to weird errors. I'd post-pone any serious refactoring once the plugin is extracted out of core.